site/src/pages/TasksPage/TasksPage.tsx

```diff
@@ -292,32 +330,83 @@ const TaskForm: FC<TaskFormProps> = ({ templates, onSuccess }) => {
 					required
 					id="prompt"
 					name="prompt"
+					value={presetAIPrompt || undefined}
+					readOnly={isPromptReadOnly}
 					placeholder={textareaPlaceholder}
 					className={`border-0 resize-none w-full h-full bg-transparent rounded-lg outline-none flex min-h-[60px]
-						text-sm shadow-sm text-content-primary placeholder:text-content-secondary md:text-sm`}
+						text-sm shadow-sm text-content-primary placeholder:text-content-secondary md:text-sm ${
+							isPromptReadOnly ? "opacity-60 cursor-not-allowed" : ""
+						}`}
 				/>
 				<div className="flex items-center justify-between pt-2">
-					<Select
-						name="templateID"
-						onValueChange={(value) => setSelectedTemplateId(value)}
-						defaultValue={templates[0].id}
-						required
-					>
-						<SelectTrigger className="w-52 text-xs [&_svg]:size-icon-xs border-0 bg-surface-secondary h-8 px-3">
-							<SelectValue placeholder="Select a template" />
-						</SelectTrigger>
-						<SelectContent>
-							{templates.map((template) => {
-								return (
-									<SelectItem value={template.id} key={template.id}>
-										<span className="overflow-hidden text-ellipsis block">
-											{template.display_name || template.name}
-										</span>
-									</SelectItem>
-								);
-							})}
-						</SelectContent>
-					</Select>
+					<div className="flex items-center gap-4">
+						<div className="flex flex-col gap-1">
+							<label
+								htmlFor="templateID"
+								className="text-xs text-content-secondary"
+							>
+								Template
+							</label>
+							<Select
+								name="templateID"
+								onValueChange={(value) => setSelectedTemplateId(value)}
+								defaultValue={templates[0].id}
+								required
+							>
+								<SelectTrigger
+									id="templateID"
+									className="w-80 text-xs [&_svg]:size-icon-xs border-0 bg-surface-secondary h-8 px-3"
+								>
+									<SelectValue placeholder="Select a template" />
+								</SelectTrigger>
+								<SelectContent>
+									{templates.map((template) => {
+										return (
+											<SelectItem value={template.id} key={template.id}>
+												<span className="overflow-hidden text-ellipsis block">
+													{template.display_name || template.name}
+												</span>
+											</SelectItem>
+										);
+									})}
+								</SelectContent>
+							</Select>
+						</div>
+
+						{presets && presets.length > 0 && (
+							<div className="flex flex-col gap-1">
+								<label
+									htmlFor="presetID"
+									className="text-xs text-content-secondary"
```

Comment:
```suggestion
									className="text-xs font-medium text-content-primary"
```

This would be a closer match to the styling elsewhere in the app